crates/ruff_python_semantic/src/cfg/graph.rs

```diff
@@ -0,0 +1,282 @@
+use ruff_index::{newtype_index, IndexVec};
+use ruff_python_ast::Stmt;
+use ruff_text_size::{Ranged, TextRange};
+use smallvec::{smallvec, SmallVec};
+
+/// Returns the control flow graph associated to an array of statements
+pub fn build_cfg(stmts: &[Stmt]) -> CFG<'_> {
+    let mut builder = CFGBuilder::with_capacity(stmts.len());
+    builder.process_stmts(stmts);
+    builder.finish()
+}
+
+/// Control flow graph
+#[derive(Debug)]
+pub struct CFG<'stmt> {
+    /// Basic blocks - the nodes of the control flow graph
+    blocks: IndexVec<BlockId, BlockData<'stmt>>,
+    /// Entry point to the control flow graph
+    initial: BlockId,
+    /// Terminal node - will always be empty
```

Comment:
Can you explain what `empty` means (as part of the comment)? Is it that the `BlockId` is empty or is it that the block is empty? 

It might be sufficient to always speak of blocks instead of nodes here (it's clear to me what an empty block is, I don't know what an empty node is)